adb_graphics/figure_builders.py

```diff
@@ -0,0 +1,280 @@
+# pylint: disable=invalid-name
+'''
+This module is where pieces of the figures are put together. Data is
+compbined with maps and skewts to provide the final product.
+'''
+
+import gc
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from adb_graphics.datahandler import gribfile
+from adb_graphics.datahandler import gribdata
+import adb_graphics.errors as errors
+from adb_graphics.figures import maps
+from adb_graphics.figures import skewt
+
+AIRPORTS = 'static/Airports_locs.txt'
+
+def add_obs_panel(ax, model_name, obs_file, proj_info, short_name, tile):
+
+    # pylint: disable=too-many-arguments
+    ''' Plot observation data provided by the obs_file
+    path using the assigned projection. '''
+
+    gribobs = gribfile.GribFile(filename=obs_file)
+    ax.axis('on')
+    field = gribdata.fieldData(
+        ds=gribobs.contents,
+        fhr=0,
+        level='obs',
+        model='obs',
+        short_name=short_name,
+        )
+    map_fields = maps.MapFields(main_field=field)
+    m = maps.Map(
+        airport_fn=AIRPORTS,
+        ax=ax,
+        grid_info=proj_info,
+        model='obs',
+        tile=tile,
+        )
+    dm = maps.MultiPanelDataMap(
+        map_fields=map_fields,
+        map_=m,
+        member='obs',
+        model_name=model_name,
+        )
+
+    # Draw the map
+    dm.draw(show=True)
+
+def parallel_maps(cla, fhr, ds, level, model, spec, variable, workdir,
+                  tile='full'):
+
+    # pylint: disable=too-many-arguments,too-many-locals
+
+    '''
+    Function that creates plan-view maps, either a single panel, or
+    multipanel for a forecast ensemble. Can be used in parallel.
+
+    Input:
+
+      fhr        forecast hour
+      ds         xarray dataset from the grib file
+      level      the vertical level of the variable to be plotted
+                 corresponding to a key in the specs file
+      model      model name: rap, hrrr, hrrre, rrfs, rtma
+      spec       the dictionary of specifications for the given variable
+                 and level
+      variable   the name of the variable section in the specs file
+      workdir    output directory
+
+    Optional:
+      tile       the label of the tile being plotted
+    '''
+
+    fig, axes = set_figure(cla.model_name, cla.graphic_type, tile)
+
+    # set last_panel to send into DataMap for colorbar control
+    last_panel = False
+
+    # Declare the type of object depending on graphic type
+    map_class = maps.MultiPanelDataMap if cla.graphic_type == \
+        'enspanel' else maps.DataMap
+
+    for index, current_ax in enumerate(axes):
```

Comment:
This loop allows us to plot 1 or many panels consistent with the figure that is setup above.